src/psyclone/psyir/transformations/remove_ifblock_trans.py

```diff
@@ -0,0 +1,145 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2022-2024, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: H. Brunie, University of Grenoble Alpes
+
+"""Module providing a transformation that removes IfBlock if the condition is
+known to be constant for the whole runtime execution."""
+
+from psyclone.psyGen import Transformation
+from psyclone.psyir.nodes import IfBlock, Routine, Node
+from psyclone.psyir.symbols import ScalarType
+from psyclone.psyir.transformations.transformation_error import (
+    TransformationError,
+)
+
+from typing import Dict, Optional
+
+
+class RemoveIfBlockTrans(Transformation):
+
+    def __init__(self, json_file_abspath: Optional[str] = None) -> None:
+        super().__init__()
+        self._known_reference_bool: Dict[str, bool] = {}
+        self._known_reference_int: Dict[str, bool] = {}
+        if json_file_abspath is not None:
```

Comment:
This part would need to either be removed entirely (and the transformation purely on PSyIR information for determining compile-time constants) or replaced with functionality that enables a list of names/symbols to be provided to the `apply` method (to support run-time constants).